language/move-lang/src/lib.rs

```diff
@@ -93,53 +129,52 @@ pub fn move_compile(
-        Ok(compiled_units) => Ok((files, compiled_units)),
-    }
+) -> anyhow::Result<(FilesSourceText, Result<Vec<CompiledUnit>, Errors>)> {
+    let (files, pprog_and_comments_res) =
+        move_parse(targets, deps, sender_opt, interface_files_dir_opt)?;
+    let (_comments, sender_opt, pprog) = match pprog_and_comments_res {
+        Err(errors) => return Ok((files, Err(errors))),
+        Ok(res) => res,
+    };
+    let result = match move_continue_up_to(PassResult::Parser(sender_opt, pprog), Pass::Compilation)
+    {
+        Ok(PassResult::Compilation(units)) => Ok(units),
+        Ok(_) => unreachable!(),
+        Err(errors) => Err(errors),
+    };
+    Ok((files, result))
 }
 
-/// Move compile but it returns the errors instead of reporting them to stderr
-pub fn move_compile_no_report(
+/// Similar to move_compile but it reports it's errors to stderr
+pub fn move_compile_and_report(
     targets: &[String],
     deps: &[String],
     sender_opt: Option<Address>,
     interface_files_dir_opt: Option<String>,
-) -> anyhow::Result<(FilesSourceText, Result<Vec<CompiledUnit>, Errors>)> {
-    let mut deps = deps.to_vec();
-    generate_interface_files_for_deps(&mut deps, interface_files_dir_opt)?;
-    let (files, pprog_and_comments_res) = parse_program(targets, &deps)?;
-    let pprog_res = pprog_and_comments_res.map(|(pprog, _)| pprog);
-    Ok(match compile_program(pprog_res, sender_opt) {
-        Err(errors) => (files, Err(errors)),
-        Ok(units) => (files, Ok(units)),
-    })
+) -> anyhow::Result<(FilesSourceText, Vec<CompiledUnit>)> {
+    let (files, units_res) = move_compile(targets, deps, sender_opt, interface_files_dir_opt)?;
+    let units = unwrap_or_report_errors!(files, units_res);
+    Ok((files, units))
 }
 
-/// Move compile up to expansion phase, returning errors instead of reporting them to stderr.
-///
-/// This also returns a map containing documentation comments for each source in `targets`.
-pub fn move_compile_to_expansion_no_report(
+pub fn move_parse(
     targets: &[String],
     deps: &[String],
     sender_opt: Option<Address>,
     interface_files_dir_opt: Option<String>,
 ) -> anyhow::Result<(
     FilesSourceText,
-    Result<(expansion::ast::Program, CommentMap), Errors>,
+    Result<(CommentMap, Option<Address>, parser::ast::Program), Errors>,
 )> {
     let mut deps = deps.to_vec();
     generate_interface_files_for_deps(&mut deps, interface_files_dir_opt)?;
     let (files, pprog_and_comments_res) = parse_program(targets, &deps)?;
-    let res = pprog_and_comments_res.and_then(|(pprog, comment_map)| {
-        let (eprog, errors) = expansion::translate::program(pprog, sender_opt);
-        check_errors(errors)?;
-        Ok((eprog, comment_map))
-    });
-    Ok((files, res))
+    let result = pprog_and_comments_res.map(|(pprog, comments)| (comments, sender_opt, pprog));
+    Ok((files, result))
+}
+
+pub fn move_continue_up_to(pass: PassResult, until: Pass) -> Result<PassResult, Errors> {
```

Comment:
Nit: comment explaining that `until` is exclusive (i.e., the `until` pass will *not* be run)?

Reply:
Other way around, it is inclusive, but I will add a comment